compiler/il/symbol/OMRResolvedMethodSymbol.cpp

```diff
@@ -579,11 +582,54 @@ OMR::ResolvedMethodSymbol::genInduceOSRCallNode(TR::TreeTop* insertionPoint,
    return insertionPoint->getPrevTreeTop();
    }
 
+/*
+ *  \brief Generate the helper call to request for OSR transition and recompilation of the current compiled method.
+ *
+ *  \return true if the OSR transition is supported at the requested \parm induceBCI and the OSR induction helper call
+ *          is generated successfully
+ */
+bool
+OMR::ResolvedMethodSymbol::induceOSRAfterAndRecompile(TR::TreeTop *insertionPoint, TR_ByteCodeInfo induceBCI, TR::TreeTop* branch,
+    bool extendRemainder, int32_t offset, TR::TreeTop ** lastTreeTop)
+   {
+   TR_ASSERT(self()->comp()->allowRecompilation(), "request OSR and recompilation at node %p when recomp is disabled\n", insertionPoint);
+   TR::TreeTop *induceOSRCallTree = induceOSRAfterImpl(insertionPoint, induceBCI, branch, extendRemainder, offset, lastTreeTop);
+   if (!induceOSRCallTree)
+      return false;
+   TR::Node *induceOSRCallNode = induceOSRCallTree->getNode()->getFirstChild();
+   TR::SymbolReference *symRef = induceOSRCallNode->getSymbolReference();
+   TR_ASSERT_FATAL(induceOSRCallNode->getOpCode().isCall() &&
+                   symRef->getReferenceNumber() == TR_induceOSRAtCurrentPC,
+                   "induceOSRCallNode %p under induceOSRCallTree %p should be a call node with TR_induceOSRAtCurrentPC helper call", induceOSRCallNode, induceOSRCallTree);
```

Comment:
To help with debugging, would be useful to also print the node id for `induceOSRCallNode`?

Reply:
added id for `induceOSRCallNode` into assert msg. Thanks!